vm/src/function.rs

```diff
@@ -328,6 +332,34 @@ impl<T> From<Vec<T>> for Args<T> {
     }
 }
 
+impl From<()> for Args<PyObjectRef> {
+    fn from(_args: ()) -> Self {
+        Args(Vec::new())
+    }
+}
+
+// A tuple of values that each implement `IntoPyObject` represents a sequence of
+// arguments that can be bound and passed to a built-in function.
+macro_rules! func_args_from_tuple {
+    ($(($n:tt, $T:ident)),*) => {
+        impl<$($T,)*> From<($($T,)* &VirtualMachine)> for Args<PyObjectRef>
```

Comment:
What about just a `IntoArgs` trait? With a `fn into_args(self, vm)`; that way you wouldn't have to pass `vm` twice in a `invoke` call (as the `self` arg to invoke, and in the argument list)

Reply:
I created IntoFuncArgs trait